src/components/HOCs/WithNearbyTasks/WithNearbyTasks.jsx

```diff
@@ -61,68 +66,141 @@ export const WithNearbyTasks = function (WrappedComponent) {
+      const isVirtual = this.isVirtualChallenge(this.props);
+      const excludeSelfLockedTasks = !!this.props.excludeSelfLockedTasks;
+
+      if (Number.isFinite(challengeId) && this.props.fetchNearbyTasks) {
+        try {
+          const nearbyTasks = await this.props.fetchNearbyTasks(
             challengeId,
             isVirtual,
-            props.taskId,
+            this.props.taskId,
             excludeSelfLockedTasks,
-            this.state.taskLimit,
-          )
-          .then((nearbyTasks) => {
-            const tasksLength = nearbyTasks.tasks.length;
-            this.setState({
-              nearbyTasks: { ...nearbyTasks, nearTaskId: props.taskId, loading: false },
-              lastLoadLength: tasksLength,
-              hasMoreToLoad: this.state.lastLoadLength !== tasksLength,
-            });
+            this.state.taskLimit < MAX_NEARBY_TASK_LIMIT
+              ? this.state.taskLimit
+              : MAX_NEARBY_TASK_LIMIT,
+          );
+
+          const tasksLength = nearbyTasks.tasks.length;
+          this.setState({
+            nearbyTasks: {
+              ...nearbyTasks,
+              nearTaskId: this.props.taskId,
+            },
+            lastLoadLength: tasksLength,
+            taskLimit: this.state.taskLimit + 5,
+            loadByNearbyTasks: true,
+            lastLoadLength: tasksLength,
+            hasMoreToLoad: this.state.lastLoadLength !== tasksLength,
           });
+        } catch (error) {
+          console.error("Error fetching nearby tasks:", error);
+        }
       }
     };
 
-    componentDidMount() {
-      this.updateNearbyTasks(this.props);
-    }
-
-    componentDidUpdate(prevProps, prevState) {
+    setMapBounds = (bounds) => {
+      // Only update if bounds have actually changed
       if (
-        this.state.nearbyTasks &&
-        !this.state.nearbyTasks.loading &&
-        this.props.taskId !== this.state.nearbyTasks.nearTaskId
+        !this.state.mapBounds ||
+        bounds.getNorth() !== this.state.mapBounds.getNorth() ||
+        bounds.getSouth() !== this.state.mapBounds.getSouth() ||
+        bounds.getEast() !== this.state.mapBounds.getEast() ||
+        bounds.getWest() !== this.state.mapBounds.getWest()
       ) {
-        this.updateNearbyTasks(this.props);
-      } else if (this.state.taskLimit !== prevState.taskLimit) {
-        this.updateNearbyTasks(this.props);
+        this.setState({ mapBounds: bounds });
       }
+    };
+
+    loadTasksInView = async () => {
+      if (!this.state.mapBounds) return;
+      const challengeId = this.currentChallengeId(this.props);
+      const isVirtual = this.isVirtualChallenge(this.props);
+      const excludeSelfLockedTasks = !!this.props.excludeSelfLockedTasks;
+
+      const bounds = this.state.mapBounds;
+      const boundingBox = [
+        bounds.getEast(),
+        bounds.getSouth(),
+        bounds.getWest(),
+        bounds.getNorth(),
+      ];
```

Comment:
This looks like it's in the wrong order. W/S/E/N is the normal order, and it looks like that's what's expected in `fetchNearbyTasksInBoundingBox()` (where `[0, 1, 2, 3]` are mapped to `left, bottom, right, top`)